test/src/state_machine/state.test.cpp

```diff
@@ -0,0 +1,771 @@
+     * in the future some values are not included in the chosen intervals,
+     * this function should be updated.
+     */
+
+    // Generates a voltage data between 100 and 399. (exact: 175 and 294)
+    battery_data.voltage = static_cast<uint16_t>((rand() % 300) + 100);
+
+    // Generates a current data between -100 and 899. (exact: 0 and 500)
+    battery_data.current = static_cast<int16_t>((rand() % 800) - 100);
+
+    // Generates a charge percentage data between 0 and 100. (exact: 20 and 100)
+    battery_data.charge = static_cast<uint8_t>(rand() % 101);
+
+    // Generates an average temperature data between 0 and 99 C. (exact: 10 and 60)
+    battery_data.average_temperature = static_cast<int8_t>(rand() % 100);
+
+    // Generates a cell voltage data between 500 and 2499 (exact: 1000 and 1296)
+    for (int i = 0; i < 36; i++) {
+      battery_data.cell_voltage[i] = static_cast<uint16_t>((rand() % 2000) + 500);
+    }
+
+    // Generates a low temperature data between 0 and 99 C. (exact: 10 and 65)
+    battery_data.low_temperature = static_cast<int8_t>(rand() % 100);
+
+    // Generates a high temperature data between 0 and 99 C. (exact: 10 and 65)
+    battery_data.high_temperature = static_cast<int8_t>(rand() % 100);
+
+    // Generates a low cell voltage data between 500 and 2499. (exact: 1000 and 1296)
+    battery_data.low_voltage_cell = static_cast<uint16_t>((rand() % 2000) + 500);
+
+    // Generates a high cell voltage data between 500 and 2499. (exact: 1000 and 1296)
+    battery_data.high_voltage_cell = static_cast<uint16_t>((rand() % 2000) + 500);
+
+    // Generates a random bool value for IMD fault.
+    battery_data.imd_fault = static_cast<bool>(rand() > (RAND_MAX / 2));
+  }
+
+  static void randomiseBatteriesData(Batteries &batteries_data)
+  {
+    randomiseModuleStatus(batteries_data.module_status);
+
+    for (auto &battery_data : batteries_data.high_power_batteries) {
+      randomiseBatteryData(battery_data);
+    }
+    for (auto &battery_data : batteries_data.low_power_batteries) {
+      randomiseBatteryData(battery_data);
+    }
+  }
+
+  //---------------------------------------------------------------------------
+  // Emergency Brakes data
+  //---------------------------------------------------------------------------
+
+  static void randomiseEmbrakes(EmergencyBrakes &embrakes_data)
+  {
+    randomiseModuleStatus(embrakes_data.module_status);
+
+    for (int i = 0; i < embrakes_data.kNumEmbrakes; i++) {
+      embrakes_data.brakes_retracted[i] = static_cast<bool>(rand() > (RAND_MAX / 2));
+    }
+  }
+
+  //---------------------------------------------------------------------------
+  // Motor data
+  //---------------------------------------------------------------------------
+
+  static void randomiseMotors(Motors &motors_data)
+  {
+    randomiseModuleStatus(motors_data.module_status);
+
+    // Generates a RPM data between 0 and 199 for all 4 motors.
```

Comment:
According to the calculations checked by propulsion, the range of rpm's should be between 33 and 1400, so the code entry is (rand() % 1368 + 33);

Reply:
I've now set it to [0, 2000] which includes the given range. Cheers